Sources/Core/Extensions/Foundation/Collection+PovioKit.swift

```diff
@@ -21,3 +21,18 @@ public extension Collection {
     lazy.filter(clause).count
   }
 }
+
+public extension Collection where Element: Dated {
+  /// Groups dated collection elements returning a dictionary
+  func grouped(by dateComponents: Set<Calendar.Component> = [.year, .month, .day]) -> [Date: [Element]] {
+    let initial: [Date: [Element]] = [:]
+    let groupedByDateComponents = reduce(into: initial) { acc, cur in
+      let components = Calendar.current.dateComponents(dateComponents, from: cur.date)
```

Comment:
I believe we need to inject the Calendar as a parameter, default value can be .current. What do you think?

Reply:
Yes, I agree 👍